server/shadow/shadow_client.c

```diff
@@ -95,9 +95,7 @@ BOOL shadow_client_context_new(freerdp_peer* peer, rdpShadowClient* client)
 	if (!(client->encoder = shadow_encoder_new(client)))
 		goto fail_encoder_new;
 
-	ArrayList_Add(server->clients, (void*) client);
-
-	return TRUE;
+	return ArrayList_Add(server->clients, (void*) client) >= 0;
```

Comment:
1. Shouldn't we do the clean-up in that case ?
2. I just noticed that the code in line 101 below is wrong.  Under the fail_encoder_new: label CloseHandle should be called with client->StopEvent and not with client->encoder !